modules/transactions.js

```diff
@@ -73,39 +73,90 @@ __private.attachApi = function () {
 
 __private.list = function (filter, cb) {
 	var sortFields = sql.sortFields;
-	var params = {}, where = [], owner = '';
+	var params = {},
+		where = [],
+		err,
+		owner = '',
+		allowedFieldsMap = {
+			blockId:             '"t_blockId" = ${blockId}',
+			senderPublicKey:     '"t_senderPublicKey" = DECODE (${senderPublicKey}, \'hex\')',
+			recipientPublicKey:  '"m_recipientPublicKey" = DECODE (${recipientPublicKey}, \'hex\')',
+			senderId:            '"t_senderId" = ${senderId}',
+			recipientId:         '"t_recipientId" = ${recipientId}',
+			fromHeight:          '"b_height" >= ${fromHeight}',
+			toHeight:            '"b_height" <= ${toHeight}',
+			fromTimestamp:       '"t_timestamp" >= ${fromTimestamp}',
+			toTimestamp:         '"t_timestamp" <= ${toTimestamp}',
+			senderIds:           '"t_senderId" IN (${senderIds:csv})',
+			recipientIds:        '"t_recipientId" IN (${recipientIds:csv})',
+			senderPublicKeys:    'ENCODE ("t_senderPublicKey", \'hex\') IN (${senderPublicKeys:csv})',
+			recipientPublicKeys: 'ENCODE ("m_recipientPublicKey", \'hex\') IN (${recipientPublicKeys:csv})',
+			minAmount:           '"t_amount" >= ${minAmount}',
+			maxAmount:           '"t_amount" <= ${minAmount}',
+			type:                '"t_type" = ${type}',
+			minConfirmations:    'confirmations >= ${minConfirmations}',
+			limit: null,
+			offset: null,
+			orderBy: null,
+			// FIXME: Backward compatibility, should be removed after transitional period
+			ownerAddress: null,
+			ownerPublicKey: null
+		};
+
+	// Generate list of fields with conditions
+	var isFirst = true;
+	_.each (filter, function (value, key) {
```

Comment:
I think it is better to stick with one syntax

https://github.com/airbnb/javascript/tree/es5-deprecated/es5#functions